### PR TITLE
[API V1 docs] Podcast Episodes

### LINKED ISSUE #18746: [API v1 docs] Podcast Episodes
We are using [rswag](https://github.com/rswag/rswag) to implement API V1 docs and are missing the following Podcast Episodes endpoints:

- [ ] [Published podcast episodes](https://developers.forem.com/api/v0#tag/podcast-episodes/operation/getPodcastEpisodes)

rswag files are added as `spec/requests/api/v1/docs/<file>_spec.rb` files. Check out [this](https://github.com/forem/forem/pull/18797) example if you need inspiration on how these look like.

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)
- [x] Documentation Update

## Description
API V1 docs for podcast episodes endpoints with [rswag](https://github.com/rswag/rswag) 

## Related Tickets & Documents
- Closes  #18746

## QA Instructions, Screenshots, Recordings
Run the specs

## Added/updated tests?

- [x] Yes

## [Forem core team only] How will this change be communicated?
- [x] I will share this change internally with the appropriate teams

### CODE REVIEW

File: spec/requests/api/v1/docs/podcast_episodes_spec.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        response "200", "A List of Podcast episodes" do
+          let(:"api-key") { nil }
+          schema  type: :array,
+                  items: { "$ref": "#/components/schemas/PodcastEpisodeIndex" }
+          add_examples
+
+          run_test!
+        end
+
+        response "200", "A List of Podcast episodes with username" do
```
Comment (Ridhwana) [inline spec/requests/api/v1/docs/podcast_episodes_spec.rb:44]: (nitpick/suggestion): Should this maybe read?

```suggestion
        response "200", "A List of Podcast episodes filtered by username" do
```

File: spec/requests/api/v1/docs/podcast_episodes_spec.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
+require "rails_helper"
+require "swagger_helper"
+
+# rubocop:disable RSpec/EmptyExampleGroup
+# rubocop:disable RSpec/VariableName
+
+RSpec.describe "Api::V1::Docs::PodcastEpisodes" do
+  # let(:api_secret) { create(:api_secret) }
+  # let(:user) { api_secret.user }
```
Comment (Ridhwana) [inline spec/requests/api/v1/docs/podcast_episodes_spec.rb:9]: (nitpick/suggestion): Should we delete these?

File: spec/swagger_helper.rb [hunk 3]
```diff
@@ -227,6 +241,15 @@
               profile_image: { description: "Profile image (640x640)", type: :string, format: :url },
               profile_image_90: { description: "Profile image (90x90)", type: :string, format: :url }
             }
+          },
+          SharedPodcast: {
+            description: "The podcast the resource belongs to",
```
Comment (Ridhwana) [inline spec/swagger_helper.rb:246]: (nitpick/suggestion): 

```suggestion
            description: "The podcast that the resource belongs to",
```

**Review by Ridhwana — APPROVED** [review]
This looks good @lightalloy ✨ I left some nitpick comments/suggestions, but nothing that is blocking.

**Review by fdocr — APPROVED** [review]
Also looks good to me overall and +1 to the inline suggestions 👍🏼

**Pull request merged**